### PR TITLE
fix(ngSrcset): ignore undefined ng-srcset directive

### LINKED ISSUE #14470: "TypeError: Cannot read property 'split' of undefined" when using ng-attr-srcset with potentially null value
**Do you want to request a *feature* or report a *bug*?**
Bug report.

**What is the current behavior?**
We're seeing an internal Angular error:

	TypeError: Cannot read property 'split' of undefined

when trying to use ng-attr-srcset, where the expression can be potentially undefined.   To reproduce,:

        

**What is the expected behavior?**

Normally, ng-attr-defined attributes are supposed to be ignored if the associated value is undefined, but in this case, we're hitting an error:

        // split srcset into tuple of uri and descriptor except for the last item
        var rawUris = trimmedSrcset.split(pattern);

where trimmedSrcset is undefined under the above situation.

**What is the motivation / use case for changing the behavior?**

ng-attr is breaking with regards to "srcset".  It should be consistent with ng-attr's behavior of undefined for other attribute names.  What's happening is an internal error that should be fixed, since it's in the framework rather than application code.

**Which versions of Angular, and which browser / OS are affected by this issue? Did this work in previous versions of Angular? Please also test with the latest stable and snapshot (https://code.angularjs.org/snapshot/) versions.**

Angular 1.5.

### PR DESCRIPTION
**Pull request by rphv**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
This is a bug fix.

**What is the current behavior? (You can also link to an open issue here)**
Currently, a TypeError is thrown when data-ng-attr-srcset is undefined.

**What is the new behavior (if this is a feature change)?**
Now, the attribute is ignored if it's undefined.

**Does this PR introduce a breaking change?**
No.

**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

Closes #14470

### CODE REVIEW

File: test/ng/directive/ngSrcsetSpec.js [hunk 1]
```diff
@@ -28,5 +28,12 @@ describe('ngSrcset', function() {
     $rootScope.$digest();
     expect(element.attr('srcset')).toBe('http://example.com/image1.png 1x,unsafe:javascript:doEvilStuff() 2x');
   }));
+
+  it('should not throw an error if undefined', inject(function($rootScope, $compile) {
```
Comment (gkalpak) [inline test/ng/directive/ngSrcsetSpec.js]: I didn't try it, but I am pretty sure this test passes without the fix.
Can you confirm that's inded tha case and change the test, so it fails without the fix ?

**Comment by gkalpak** [comment]
Thx @rphv for working on this.
Could you address my comment ? (Feel free to ping me once you have, so I can take another look.)

File: test/ng/directive/ngSrcsetSpec.js (hunk 1 above)
Comment (rphv) [inline test/ng/directive/ngSrcsetSpec.js]: Yes, you're right. That test wasn't testing the fix.

I've updated the test and verified that it fails without the fix.

**Comment by rphv** [comment]
Test updated.

On Fri, Apr 22, 2016 at 3:57 AM, Georgios Kalpakas <notifications@github.com

**Comment by gkalpak** [comment]
@rphv, since this is not `ng-attr-` specific (and there is no guarantee that its implementatation will keep exercising the same code paths in the future), could you also add a test that call `attrs.$set('srcset', undefined)` directly on an `` element ?

(Also, you don't have to use `data-` prefixes in tests - they are just adding clutter.)

**Comment by rphv** [comment]
Thank you for your advice and attention.

I'm not sure if the new test I added is in the right place or in the
accepted style of the project. However, it does fail without the fix.

**Comment by gkalpak** [comment]
Tweaked the test a bit and merged.
Thx for working on this :+1:

**Pull request closed without merging**